platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/registry/PrometheusPushMeterRegistry.java

```diff
@@ -0,0 +1,86 @@
+package org.hypertrace.core.serviceframework.metrics.registry;
+
+import io.micrometer.core.instrument.util.TimeUtils;
+import io.micrometer.core.lang.Nullable;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.prometheus.client.exporter.PushGateway;
+import java.io.IOException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import org.hypertrace.core.serviceframework.metrics.PlatformMetricsRegistry;
+import org.hypertrace.core.serviceframework.metrics.config.PrometheusPushRegistryConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Metric Registry for Prometheus Push Gateway
+ */
+public class PrometheusPushMeterRegistry extends PrometheusMeterRegistry {
+  private static final Logger logger = LoggerFactory.getLogger(PlatformMetricsRegistry.class);
+  private final PrometheusPushRegistryConfig pushConfig;
+  private final PushGateway pushGateway;
+
+  @Nullable
+  private ScheduledExecutorService scheduledExecutorService;
+
+  public PrometheusPushMeterRegistry(
+      PrometheusPushRegistryConfig pushConfig, ThreadFactory threadFactory, PushGateway pushGateway) {
```

Comment:
If we are working with ScheduledExecutorService created through Executors, that would by-default create threadFactory internally through `Executors.defaultThreadFactory()`. We don't need to pass on the same as a separate argument.

Reply:
Easier to test because then we can mock it. 
Also, make it consistent with PushMetricRegistry.